### PR TITLE
MortarGrid sides are identified by Enum

### PR DESCRIPTION
**Pull request by keileg**
This leaves a more robust implementation than previously, when we relied on accessing module-level constants outside the module.

Also, added tests to check pickling of `Grid`, `MortarGrid` and `GridBucket`.

### CODE REVIEW

File: src/porepy/grids/mortar_grid.py [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
-NONE_SIDE = 0
-LEFT_SIDE = 1
-RIGHT_SIDE = 2
-WHOLE_SIDE = np.iinfo(type(NONE_SIDE)).max
+
+class MortarSides(Enum):
+    # Enum of constants used to identify the grids on each side of the mortar
+    NONE_SIDE = 0
+    LEFT_SIDE = 1
+    RIGHT_SIDE = 2
```
Comment (alessiofumagalli) [inline src/porepy/grids/mortar_grid.py:17]: why did you remove the WHOLE_SIDE? We probably never used it, but maybe you've a specific reason

**Review by alessiofumagalli — APPROVED** [review]
It looks good, thanks @keileg

File: src/porepy/grids/mortar_grid.py (hunk 1 above)
Comment (keileg) [reply src/porepy/grids/mortar_grid.py:17]: 1) I did not understand when we need it, so we'll introduce it later if necessary. 
2) If we reintroduce it, it should get a standard number, not some strange trick to get a random number.
